app/controllers/items/detail.js

```diff
@@ -294,6 +294,22 @@ export default GoodcityController.extend(
       }
     },
 
+    onSaleableChange: Ember.observer("item.saleable", function() {
```

Comment:
A few things:

* An observer for a checkbox feels a bit too agressive. I think having a "saveItem" action linked to the checkbox's `change` event is more appropriate.
* Having a `previousSaleableValue` value doesn't seem used in any meaningful way here. If there is an error you can do `item. rollbackAttributes()` to restore it as it was (I assume that's what you were going for)

Also, we have an "async" mixin you can use to avoid having to take care of the loading spinner.

e.g

```javascript
{
   saveItem() {
      const item = this.get("item");
      this.runTask(async () => {
           try {
               await item.save();
           } catch (e) {
               item.rollbackAttributes();
               throw e;
           }
      }, this.ERROR_STRATEGIES.MODAL);
  }
}

// hbs 
{{input type="checkbox" checked=isSaleable change=(action "saveItem") }} // might be wrong about the syntax
```

Reply:
thanks @patrixr ,
`runTask` was awesome sleek. I have updated it, as well as created action instead of observers